### PR TITLE
Update DPF to pull in fix for plugin GUI on some hosts.

### PR DESCRIPTION
**Pull request by Justin-Hoffman**
This updates the DPF commit hash to a recent develop branch version. 

It fixes a crash in the plugin GUI that was being experienced on some hosts.

### CODE REVIEW

**Comment by falkTX** [comment]
the way to enable file browser is very hacky, we just need an optional arg to `dpf_add_plugin`
see https://github.com/DISTRHO/DPF/blob/develop/cmake/DPF-plugin.cmake#L105

**Comment by Justin-Hoffman** [comment]
> the way to enable file browser is very hacky, we just need an optional arg to `dpf_add_plugin` see https://github.com/DISTRHO/DPF/blob/develop/cmake/DPF-plugin.cmake#L105

Copy that. Will fix.

**Comment by Justin-Hoffman** [comment]
Is the runner label "macos-11" retired? That job has been waiting for a runner for over 8 hours and I only see 12+ here: 
https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
     strategy:
       matrix:
         target: [macos-universal-10.15]
-    runs-on: macos-11
+    runs-on: macos-12
```
Comment (falkTX) [inline .github/workflows/build.yml:28]: this needs to be 13, github and apple and deprecating old versions quite fast

Comment (Justin-Hoffman) [reply .github/workflows/build.yml]: 13 it is.

**Comment by Justin-Hoffman** [comment]
The Mac builds hit the error that I had seen on Linux with an older DPF version: 
/Users/runner/work/AIDA-X/AIDA-X/src/jucewrapper/../../modules/dpf/distrho/src/DistrhoPluginChecks.h:167:3: error: invalid build config: file browser requested but `USE_FILE_BROWSER` build option is not set
# error invalid build config: file browser requested but `USE_FILE_BROWSER` build option is not set

I'm guessing for that platform we need to do something different? Any suggestions here @falkTX ?

**Comment by falkTX** [comment]
> Yeah, I can trigger the same error locally if I enable the JUCE build here (which is only on by default for Mac, hence the build issue only for that build. With these last fixes I can build the JUCE plugin here locally without issue.

I completely forgot I had done a JUCE wrapper just to get AU, hah.
It was a while ago, now DPF supports AU natively, so we can get rid of the JUCE hacks. but should be a separate PR, will need tweaks to some core files to implement it properly anyhow

**Comment by Justin-Hoffman** [comment]
And now we have: 
`make[2]: *** No rule to make target `bin/AIDA-X.a', needed by `bin/AIDA-X.component/Contents/MacOS/AIDA-X'.  Stop.`

Looking into it

**Comment by Justin-Hoffman** [comment]
> the entire juce thing is a big hack, so best to just ignore it or just outright remove it.

Understood. Mind approving this pipeline to see if simply specifying linked library as whatever the existing targets emit will resolve this? 

If that doesn't work, or if you're not happy with this, I can also just strip the JUCE / AU build out entirely.

**Comment by falkTX** [comment]
Sure thing! can set up the AU stuff now too

**Pull request merged**